### PR TITLE
Don't recursively scan for pyproject.toml when calculating the cache key

### PR DESCRIPTION
**Pull request by llucax**
We don't really need to scan for `pyproject.toml` files recursively, since we only have one in the root of the repository. This should make the cache key calculation more efficient and less error prone, as when using qemu, there are some files that are not accessible and the hash calculation fails.

### CODE REVIEW

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Comment by llucax** [comment]
Not sure why this says it still needs an approval from someone with write access, as @daniel-zullo-frequenz has write access :thinking:

**Comment by llucax** [comment]
I re-enabled auto-merge and it worked now :shrug:

**Pull request merged**
